packages/wrangler/src/kv.ts

```diff
@@ -205,15 +206,38 @@ export async function deleteKVKeyValue(
   );
 }
 
+/**
+ * Formatter for converting e.g. 5,328 --> 5.3K
+ */
+const compact = new Intl.NumberFormat("en-US", {
+  compactDisplay: "short",
+  notation: "compact",
+}).format;
+
+/**
+ * Helper function for bulk requests, logs ongoing output to console.
+ */
+function logBulkProgress(
+  request: "put" | "delete",
+  index: number,
+  total: number
+) {
+  logger.log(
+    `${request === "put" ? "Uploaded" : "Deleted"} ${Math.floor(
+      (100 * index) / total
+    )}% (${compact(index)} out of ${compact(total)})`
+  );
+}
+
 export async function putKVBulkKeyValue(
   accountId: string,
   namespaceId: string,
   keyValues: KeyValue[],
-  progressCallback: (index: number, total: number) => void
+  quiet = false
```

Comment:
Out of interest... why change the design so that the methods now know about logging? An alternative would be to provide a helper function that creates a progressCallback...

```ts
await putKVBulkKeyValue(
  accountId,
  namespaceId,
  content,
  logProgress("put"),
);
```

where 

```ts
function logProgress(operation: "put"|"request"): (index: number, total: number) => void {
 ...
}
```

Reply:
And then I think there would be zero changes to this file.